apps/prairielearn/elements/pl-threejs/pl-threejs.py

```diff
@@ -375,10 +373,10 @@ def render(element_html, data):
             "options": json.dumps(options, allow_nan=False),
         }
 
-        with open("pl-threejs.mustache", "r", encoding="utf-8") as f:
+        with open("pl-threejs.mustache", encoding="utf-8") as f:
             html = chevron.render(f, html_params).strip()
     else:
-        raise Exception("Invalid panel type: %s" % data["panel"])
+        raise Exception("Invalid panel type: {}".format(data["panel"]))
```

Comment:
Any idea why this wasn't changed to use an f-string instead?

Reply:
I think this would cause a nested string, which it may not be able to autofix

Reply:
From https://github.com/asottile/pyupgrade?tab=readme-ov-file#f-strings:

> note: pyupgrade is intentionally timid and will not create an f-string if it would make the expression longer or if the substitution parameters are sufficiently complicated (as this can decrease readability).